src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,134 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "ValidLogin";
+    private static final String SHORT_LOGIN = "abc";
+    private static final String VALID_PASSWORD = "Password1!";
+    private static final String INVALID_PASSWORD_NO_LOWERCASE = "PASSWORD1!";
+    private static final String INVALID_PASSWORD_NO_UPPERCASE = "password1!";
+    private static final String INVALID_PASSWORD_NO_DIGITS = "Password!";
+    private static final String INVALID_PASSWORD_NO_SPECIAL_SYMBOL = "Password1";
+    private static final String INVALID_PASSWORD_SHORT = "Pas1!";
+    private static final int VALID_AGE = 25;
+    private static final int NON_VALID_AGE = 17;
+    private RegistrationService registrationService;
+    private StorageDao storageDao;
+
+    @BeforeEach
+    public void setup() {
+        storageDao = new StorageDaoImpl();
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    public void drop() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        assertThrows(UserValidationException.class,
+                () -> registrationService.register(null));
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        User user = new User(null, VALID_PASSWORD, VALID_AGE);
+        assertThrows(UserValidationException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        User user = new User(VALID_LOGIN, null, VALID_AGE);
+        assertThrows(UserPasswordValidationException.class,
+                () -> registrationService.register(user));
+    }
+
+    @Test
+    public void testValidUserRegistration_ok() {
+        User validUser = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        User returnedUser = registrationService.register(validUser);
+        assertEquals(validUser, returnedUser);
+    }
+
+    @Test
+    public void testRegistrationWithNullUser_notOk() {
+        assertThrows(UserValidationException.class, () -> registrationService.register(null));
+    }
+
+    @Test
+    public void testRegistrationWithExistingLogin_notOk() {
+        User newUser = new User(VALID_LOGIN, VALID_PASSWORD, VALID_AGE);
+        registrationService.register(newUser);
```

Comment:
Check common mistakes, we should not arrange test data(putting already existing user to DB) via service publci methods, we should instead use more low-level ways like direct addition to `Storage.people`, or using DAO in that case. Later in will be done in more advanced ways, but for now it'll do